src/scripts/main.js

```diff
@@ -1,3 +1,37 @@
 'use strict';
 
+const bodyElement = document.body;
+const tableElement = bodyElement.querySelector('table');
+const tHead = tableElement.querySelector('thead');
+const tBody = tableElement.querySelector('tbody');
+const tFoot = tableElement.querySelector('tfoot');
+
+const trHead = tHead.querySelector('tr');
+const trBody = tBody.querySelector('tr');
+const trFoot = tFoot.querySelector('tr');
+
+const addList = (tr) => {
+  const thBody = tBody.querySelectorAll('td:nth-child(2)');
+
+  for (let i = 0; i < tr.length; i++) {
+    const row = tr[i];
+    const lastChild = row.querySelector('td:last-child');
+    const newTd = document.createElement('td');
+
+    newTd.textContent = thBody[i].textContent;
+    row.insertBefore(newTd, lastChild);
+  }
+};
+
+const addHeadFoot = (tr, elemToAdd) => {
+  const lastChild = tr.querySelector('th:last-child');
+  const newRow = document.createElement('th');
+
+  newRow.textContent = tr.querySelector(elemToAdd).textContent;
```

Comment:
Ensure that 'tr.querySelector(elemToAdd)' returns a valid element. If 'elemToAdd' does not match any element, this will cause an error.